37 add support for deploying objects in qs

bkutlu opened the pull request:
Ability to write and read data objects in qs format will improve the speed with large daaps. See issue (#94) in `dpbuild` for details.  I had to create some helper functions to check the file format before saving. Please take a look closely and test it on your environment.

amashadihossein committed:
Merge pull request #11 from amashadihossein/dev

PR to remove dp_archive

amashadihossein committed:
Merge pull request #13 from amashadihossein/dev

Update pkgdown site, remove docs, update gh actions

amashadihossein committed:
Merge pull request #16 from amashadihossein/dev

Dev

amashadihossein committed:
Merge pull request #18 from amashadihossein/dev

Dev

afshinmhCLG committed:
Merge pull request #22 from amashadihossein/dev

Dev

camorosi committed:
Merge pull request #28 from amashadihossein/dev

pins v1 release (0.1.0)

camorosi committed:
Merge pull request #32 from amashadihossein/dev

Add pinsLabkey

amashadihossein committed:
Merge pull request #36 from amashadihossein/dev

v0.3.0 fully deprecate board_alias

bkutlu committed:
added support for qs file deployment

bkutlu committed:
fixed a missing bracket

bkutlu committed:
read qs modified

bkutlu committed:
read_qs correctly returning the data object

bkutlu committed:
fixed the issue with RDS read

bkutlu committed:
added type arg to deployCore method

bkutlu committed:
document and DESCRIPTION modified

camorosi commented on R/dp_deploy.R:
```diff
@@ -22,30 +22,47 @@ dp_deploy <- function(project_path = ".", ...) {
-    "data_object.RDS"
-  ))
+  # get the data object file type
+  type <- fs::dir_ls(file.path(project_path,'output_files/'), recurse = T, regexp = 'data_object') |> 
+      tools::file_ext() |> 
```
We haven't used the native pipe anywhere else in the daapr codebase to my knowledge. Recommend keeping magrittr pipe for now for compatibility with older R versions

bkutlu committed:
added type parameter to dpinput_sync

bkutlu committed:
removed base pipe and manual added

bkutlu committed:
Added tests

camorosi commented on R/dp_deploy.R:
```diff
@@ -22,30 +22,51 @@ dp_deploy <- function(project_path = ".", ...) {
   )
 }
 
+detect_type = function(project_path){
```
As mentioned in dpbuild PR: what if daap was started as rds and switched to qs? In this case the detected type will be wrong

bkutlu replied:
Please see my reply: [here](https://github.com/amashadihossein/dpbuild/pull/99#discussion_r1887745128)

camorosi commented on R/dp_deploy.R:
```diff
@@ -22,30 +22,51 @@ dp_deploy <- function(project_path = ".", ...) {
+      tolower()
+}
+
+object_read <- function(project_path, type){
```
Can you document these internal functions and tag as noRd?

bkutlu committed:
added type param get_pin_version and added docs

camorosi committed:
Fix documentation warnings

camorosi reviewed (approved):
I tested with local and s3 data products and everything is working smoothly.

One small change I would recommend is modifying `dpboardlog_update()` to pass `type = "rds"` to the `pin_write()` calls directly to suppress the message of "Guessing type = 'rds'" for the board log. This message is confusing when writing a qs type daap. 

Note that with the current implementation daaps with multiple types (ie a daap started with rds and switched to qs) may have unexpected behavior with `detect_type()` but I think this can be handled via documentation in a future qs vignette.

bkutlu committed:
Set rds as default type for boardlog

bkutlu commented:
> I tested with local and s3 data products and everything is working smoothly.
> 
> One small change I would recommend is modifying `dpboardlog_update()` to pass `type = "rds"` to the `pin_write()` calls directly to suppress the message of "Guessing type = 'rds'" for the board log. This message is confusing when writing a qs type daap.
> 
> Note that with the current implementation daaps with multiple types (ie a daap started with rds and switched to qs) may have unexpected behavior with `detect_type()` but I think this can be handled via documentation in a future qs vignette.

Thanks, I added 'rds' as default for boardlog to avoid confusion.
